### ISSUE TITLE
pyfa Error

### ISSUE THREAD

**Issue by rosborn** [DarkFenX/Pyfa#316]
After downloading pyfa 1.13.0, this error message pops up when I open pyfa:
![screen shot 2015-07-07 at 5 02 43 pm](https://cloud.githubusercontent.com/assets/92023/8558491/30e6e2bc-24ca-11e5-8f00-1f3d5bc01413.png)
.pyfa/log.txt is empty.
I was previously running pyfa 1.12.0 without any problems.
### Software info:
```
System Version:	OS X 10.10.4 (14E46)
  Kernel Version:	Darwin 14.4.0
  Boot Volume:	Macintosh HD
  Boot Mode:	Normal

Python Version: 2.7.10
```
### Hardware info:
```
Model Name:	MacBook Pro
  Model Identifier:	MacBookPro8,2
  Processor Name:	Intel Core i7
  Processor Speed:	2.2 GHz
  Number of Processors:	1
  Total Number of Cores:	4
  L2 Cache (per Core):	256 KB
  L3 Cache:	6 MB
  Memory:	4 GB
  Boot ROM Version:	MBP81.0047.B2A
  SMC Version (system):	1.69f4
```
### Open files and ports:
```
/Applications/pyfa.app/Contents/Resources
/Applications/pyfa.app/Contents/MacOS/pyfa
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/zlib.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/strop.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/operator.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_functools.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_locale.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_weakref.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_struct.so
/Applications/pyfa.app/Contents/Frameworks/Python.framework/Versions/2.7/Python
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/time.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/cStringIO.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_collections.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/itertools.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_bisect.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_heapq.so
/usr/share/icu/icudt53l.dat
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_socket.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_ssl.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/cPickle.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/wx/_core_.so
/Applications/pyfa.app/Contents/Frameworks/libwx_macud-2.8.0.dylib
/System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ColorSync.framework/Versions/A/Resources/ColorSyncDeprecated.dylib
/System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libCGCMS.A.dylib
/System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libRIP.A.dylib
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_hashlib.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/binascii.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/wx/_gdi_.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/wx/_windows_.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/wx/_controls_.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/wx/_misc_.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_sha256.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_sha512.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/datetime.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/math.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_scproxy.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_random.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/fcntl.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/grp.so
/Applications/pyfa.app/Contents/Resources/lib/python2.7/lib-dynload/_sqlite3.so
/System/Library/ColorSync/Profiles/Generic RGB Profile.icc
/System/Library/ColorSync/Profiles/sRGB Profile.icc
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/SystemAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/GraphiteDarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityDarkGraphiteAppearance.car
/System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libCGXType.A.dylib
/private/var/folders/dm/yxjgnbys6_q03z_hx995_98r0000gn/0/com.apple.LaunchServices-107501.csstore
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/GraphiteAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityGraphiteAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/DarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityDarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/VibrantLightAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityVibrantLightAppearance.car
/System/Library/PrivateFrameworks/CloudDocs.framework/Versions/A/CloudDocs
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/Extras2.rsrc
/Applications/pyfa.app/Contents/Resources/pyfa.icns
/System/Library/ColorSync/Profiles/Generic Gray Profile.icc
/System/Library/ColorSync/Profiles/Generic Gray Gamma 2.2 Profile.icc
/System/Library/Fonts/HelveticaNeueDeskInterface.ttc
/System/Library/Fonts/Helvetica.dfont
/System/Library/PrivateFrameworks/CoreUI.framework/Versions/A/Resources/SArtFile.bin
/System/Library/Caches/com.apple.IntlDataCache.le.kbdx
/private/var/folders/dm/yxjgnbys6_q03z_hx995_98r0000gn/C/com.apple.scriptmanager2.le.cache
/System/Library/Keyboard Layouts/AppleKeyboardLayouts.bundle/Contents/Resources/AppleKeyboardLayouts-L.dat
/System/Library/Frameworks/OpenCL.framework/Versions/A/Libraries/ImageFormats/unorm8_bgra.dylib
/usr/lib/dyld
/dyld_shared_cache_i386
/dev/null
/dev/null
/dev/null
count=2, state=0x2
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/SystemAppearance.car
~/.pyfa/log.txt
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/GraphiteAppearance.car
/Applications/pyfa.app/Contents/Resources/staticdata/eve.db
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityGraphiteAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/DarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/GraphiteDarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityDarkAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityDarkGraphiteAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/VibrantLightAppearance.car
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityVibrantLightAppearance.car
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/Extras2.rsrc
~/Library/Saved Application State/org.evefit.pyfa.savedState/data.data
~/Library/Saved Application State/org.evefit.pyfa.savedState/data.data
~/Library/Saved Application State/org.evefit.pyfa.savedState/windows.plist
~/Library/Saved Application State/org.evefit.pyfa.savedState/window_1.data
```

If you need any other info let me know.

**Comment by msgonzo** [comment]
I am getting this same error.

**Comment by blitzmann** [comment]
Are you dev savvy? If so, can you run pyfa from the python interpreter? You'll need wxPython 2.8 installed as well as a few other python modules. My OS X VM isn't working right

**Comment by blitzmann** [comment]
Got my VM to (kind of) work, but my python packages are all fucked, so my errors aren't reliable. Looking into it.

**Comment by rosborn** [comment]
Sorry, I don't seem to be able to install wxpython2.8. I think it has something to do with python2.7 being in 64bit mode. Not really sure. Don't have enough time to look into it.

**Comment by blitzmann** [comment]
`ImportError: cannot import name flag_modified`

Well that is disconcerting. We are using a very old skeleton for the mac client as we do not have a mac developer to make a new one. Either this skeleton doesn't have `flag_modified` or sqlalchemy 0.6 did not have it.

Only way to properly fix this that I can see would be to update the os x build. In the short term though, we might be able to work around this using a disgusting hack. I'll have to test it out. Still not sure what's up with the `-r` flag =/

**Comment by blitzmann** [comment]
Should be fixed with v1.13.1. Please test it out and reports back. If you still have issues, reopen.

Note: I did not test `-r` flag. That may or may not work (but I suspect it will as we also fixed an issue with savePath creation)

**Comment by maurerit** [comment]
What's the build procedures?  I could probably run the build for you if you like!

**Comment by rosborn** [comment]
Everything seems to be working with the new update. Thanks for the quick response.  
@maurerit Building wxpython2.8 for the mac is quite difficult. You'll need to get a hold of the 10.6 sdk and link to that when building.
https://github.com/DarkFenX/Pyfa/wiki/FAQ#requirements-for-running-pyfa-from-source
https://forums.wxwidgets.org/viewtopic.php?t=38413#p158090
I haven't had the time to build all the pyfa requirements, so ymmv.

**Comment by blitzmann** [comment]
You shouldn't have to build wx2.8 for the mac. I got it working from the installer: http://sourceforge.net/projects/wxpython/files/wxPython/2.8.12.1/

When I say build, I mean the skeleton. The mac package is basically a skeleton of mac-specific files (see skeleton repo on DarkFenX) that the pyfa source is just copied into, and the whole thing is zipped to make the release. The problem is that it contains old libraries (sqlalchemy 0.6.4 and wx2.8), and we made years ago. We no longer have a developer with access to an actual mac, so progress on anything mac-related is slow. To build a new one requires running something like py2app or cxFreeze to update everything. We're going to need to do this anyway for the wx3 release as soon as the bugs for that are worked out. But more about the mac build efforts can be found at @289
